pkg/engine/engine.go

```diff
@@ -233,6 +234,21 @@ func (e *Engine) BytesScanned() uint64 {
 	return e.bytesScanned
 }
 
+func (e *Engine) dedupeAndSend(chunkResults []detectors.ResultWithMetadata) {
```

Comment:
I was a bit curious so I did some benchmarking with test with and without the tiny changes i mentioned ad see some pretty nice perf gains.
![Screenshot 2023-07-11 at 12 23 53 PM](https://github.com/trufflesecurity/trufflehog/assets/21311841/948320ff-14ca-4743-a501-9feb174e48ae)

Test and benchmark code if you were interested:
```
func TestDedupeAndSend(t *testing.T) {
	testCases := []struct {
		name     string
		input    []detectors.ResultWithMetadata
		expected int
	}{
		{
			name: "duplicate detection",
			input: []detectors.ResultWithMetadata{
				{
					Result: detectors.Result{
						DetectorType: detectorspb.DetectorType_AWS,
						Raw:          []byte("test"),
					},
					SourceMetadata: &source_metadatapb.MetaData{
						Data: &source_metadatapb.MetaData_Slack{
							Slack: &source_metadatapb.Slack{
								ChannelName: "test",
							},
						},
					},
				},
				{
					Result: detectors.Result{
						DetectorType: detectorspb.DetectorType_AWS,
						Raw:          []byte("test"),
					},
					SourceMetadata: &source_metadatapb.MetaData{
						Data: &source_metadatapb.MetaData_Slack{
							Slack: &source_metadatapb.Slack{
								ChannelName: "test",
							},
						},
					},
				},
			},
			expected: 1,
		},
		// Add more test cases here...
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			ch := make(chan detectors.ResultWithMetadata, len(tc.input))
			e := Engine{results: ch}

			e.dedupeAndSend(tc.input)

			if len(ch) != tc.expected {
				t.Errorf("Expected %d result(s), got %d", tc.expected, len(ch))
			}
		})
	}
}

func BenchmarkDedupeAndSend(b *testing.B) {
	ch := make(chan detectors.ResultWithMetadata, 5000)
	e := Engine{results: ch}

	results := make([]detectors.ResultWithMetadata, 5000)
	for i := 0; i < 5000; i++ {
		rawValue := fmt.Sprintf("test%d", i)

		// Insert duplicate values every 100th index
		if i%100 == 0 {
			rawValue = "duplicateValue"
		}

		results[i] = detectors.ResultWithMetadata{
			Result: detectors.Result{
				DetectorType: detectorspb.DetectorType_AWS,
				Raw:          []byte(rawValue),
			},
			SourceMetadata: &source_metadatapb.MetaData{
				Data: &source_metadatapb.MetaData_Slack{
					Slack: &source_metadatapb.Slack{
						ChannelName: "test",
					},
				},
			},
		}
	}

	// Start a goroutine to read from the channel concurrently.
	done := make(chan bool)
	go func() {
		for {
			select {
			case <-ch:
				// Read from channel and do nothing.
			case <-done:
				// If done signal is received, stop reading.
				return
			}
		}
	}()

	b.ResetTimer()

	for n := 0; n < b.N; n++ {
		e.dedupeAndSend(results)
	}

	// Send a signal to stop reading from the channel.
	done <- true
}
```